src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/OptimizedInboxTextEncoder.Ssse3.cs

```diff
@@ -0,0 +1,230 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace System.Text.Encodings.Web
+{
+    internal sealed partial class OptimizedInboxTextEncoder
+    {
+        private unsafe nuint GetIndexOfFirstByteToEncodeSsse3(byte* pData, nuint lengthInBytes)
+        {
+            Debug.Assert(Ssse3.IsSupported);
+            Debug.Assert(BitConverter.IsLittleEndian);
+
+            var vecZero = Vector128<byte>.Zero;
+            var vec0x7 = Vector128.Create((byte)0x7);
+            var vecPowersOfTwo = Vector128.Create(1, 2, 4, 8, 16, 32, 64, 128, 0, 0, 0, 0, 0, 0, 0, 0);
+            var allowedCodePoints = _allowedAsciiCodePoints.AsVector;
+            int pmovmskb;
+
+            nuint i = 0;
+            if (lengthInBytes >= 16)
+            {
+                nuint lastLegalIterationFor16CharRead = lengthInBytes & unchecked((nuint)(nint)~0xF);
+
+                do
+                {
+                    // Read 16 bytes at a time into a single 128-bit vector.
+
+                    var packed = Unsafe.ReadUnaligned<Vector128<byte>>(pData + i);
```

Comment:
Good catch - this was originally written in terms of _ref byte_ instead of _byte\*_, and I forgot to update these call sites when converting the thing to be pointer based.